feeds/npm/npm.go

```diff
@@ -117,16 +127,34 @@ func (feed Feed) Latest(cutoff time.Time) ([]*feeds.Package, error) {
 		return pkgs, err
 	}
 
+	// Handle the possibility of multiple releases of the same package
+	// within the polled `packages` slice.
+	uniquePackages := make(map[string][]*Package)
 	for _, pkg := range packages {
-		go func(pkg *Package) {
-			v, err := fetchVersionInformation(pkg.Title)
+		v, ok := uniquePackages[pkg.Title]
+		if ok {
+			v = append(v, pkg)
+			uniquePackages[pkg.Title] = v
+		} else {
+			pkgs := []*Package{}
+			pkgs = append(pkgs, pkg)
+			uniquePackages[pkg.Title] = pkgs
+		}
+	}
+
+	for _, pkgs := range uniquePackages {
+		go func(pkgs []*Package) {
+			versionedPkgs, err := fetchVersionInformation(pkgs)
 			if err != nil {
 				errs <- err
 				return
 			}
-			feedPkg := feeds.NewPackage(pkg.CreatedDate.Time, pkg.Title, v, FeedName)
-			packageChannel <- feedPkg
-		}(pkg)
+			for _, versionedPkg := range versionedPkgs {
+				feedPkg := feeds.NewPackage(versionedPkg.CreatedDate, versionedPkg.Title,
+					versionedPkg.Version, FeedName)
+				packageChannel <- feedPkg
```

Comment:
Is `len(versionedPkgs) == len(packages)`?

Reply:
If you were to total the `versionedPkgs` from each iteration then yes, unless I've made a mistake